assets/js/googlesitekit/api/cache.test.js

```diff
@@ -0,0 +1,321 @@
+					expect( result.cacheHit ).toEqual( true );
+					expect( result.value ).toEqual( undefined );
+				} );
+
+				it( 'should return a number value', async () => {
+					const didSave = await set( 'number', 500 );
+					expect( didSave ).toEqual( true );
+
+					const result = await get( 'number', 100 );
+
+					expect( storageMechanism.getItem ).toHaveBeenCalledWith( 'number' );
+					expect( result.cacheHit ).toEqual( true );
+					expect( result.value ).toEqual( 500 );
+				} );
+
+				it( 'should return an array value', async () => {
+					const didSave = await set( 'array', [ 1, '2', 3 ] );
+					expect( didSave ).toEqual( true );
+
+					const result = await get( 'array', 100 );
+
+					expect( storageMechanism.getItem ).toHaveBeenCalledWith( 'array' );
+					expect( result.cacheHit ).toEqual( true );
+					expect( result.value ).toEqual( [ 1, '2', 3 ] );
+				} );
+
+				it( 'should return an object value', async () => {
+					const didSave = await set( 'object', { foo: 'barr' } );
+					expect( didSave ).toEqual( true );
+
+					const result = await get( 'object', 100 );
+
+					expect( storageMechanism.getItem ).toHaveBeenCalledWith( 'object' );
+					expect( result.cacheHit ).toEqual( true );
+					expect( result.value ).toEqual( { foo: 'barr' } );
+				} );
+
+				it( 'should return a complex value', async () => {
+					const didSave = await set( 'complex', [ 1, '2', { cool: 'times', other: [ { time: { to: 'see' } } ] } ] );
+					expect( didSave ).toEqual( true );
+
+					const result = await get( 'complex', 100 );
+
+					expect( storageMechanism.getItem ).toHaveBeenCalledWith( 'complex' );
+					expect( result.cacheHit ).toEqual( true );
+					expect( result.value ).toEqual( [ 1, '2', { cool: 'times', other: [ { time: { to: 'see' } } ] } ] );
+				} );
+			} );
+
+			describe( 'set', () => {
+				it( 'should not save when a hard-to-serialize data is set', async () => {
+					const arrayBuffer = new ArrayBuffer( 8 );
+					const didSave = await set( 'arrayBuffer', arrayBuffer, 500 );
+					const storedData = JSON.stringify( {
+						timestamp: 500,
+						value: arrayBuffer,
+					} );
+
+					expect( didSave ).toEqual( false );
+					expect( storageMechanism.setItem ).not.toHaveBeenCalledWith( 'arrayBuffer', storedData );
+					expect( Object.keys( storageMechanism.__STORE__ ).length ).toBe( 0 );
+				} );
+
+				it( 'should save data', async () => {
+					const didSave = await set( 'array', [ 1, 2, 3 ], 500 );
+					const storedData = JSON.stringify( {
+						timestamp: 500,
+						value: [ 1, 2, 3 ],
+					} );
+
+					expect( didSave ).toEqual( true );
+					expect( storageMechanism.setItem ).toHaveBeenCalledWith( 'array', storedData );
+					expect( Object.keys( storageMechanism.__STORE__ ).length ).toBe( 1 );
+				} );
+			} );
+
+			describe( 'deleteItem', () => {
+				it( 'should delete data', async () => {
+					const didSave = await set( 'array', [ 1, 2, 3 ], 500 );
```

Comment:
Probably not, might be a bit of copypasta. I'll tidy it up, good call 👍

Reply:
Oh wait: it does. We check the entire call to `localStorage.setItem` that way, and because the timestamp is part of that `setItem` call we hardcore it so we can verify it's saved. I'll add a comment though; it's (obviously) super-unclear from this code.